elements/openlayers-elements/ol-overlay.ts

```diff
@@ -77,19 +77,32 @@ export default class OlOverlay extends OlMapPart<Overlay> {
     const slotEl = document.createElement('slot')
     slotEl.name = slot
 
-    this.__overlay = new Overlay({
+    const overlayOptions: {
+      element: HTMLElement
+      autoPan?: {
+        animation: {
+          duration: number
+        }
+      }
+    } = {
```

Comment:
This type is exported so you could `import { Options } from 'ol/Overlay.js'` and use it to annotate

```suggestion
    const overlayOptions: Options = {
```